### PR TITLE
load multiple libs in one call

### LINKED ISSUE #5344: Add ability to load multiple libs in one call
## Description
`sy.load()` should accept any valid sequence of strings, variadic args, a list, tuple or set of strings like:
```
sy.load("pydp", "opacus")
sy.load(["pydp", "opacus"])
sy.load(("pydp", "opacus"))
sy.load({"pydp", "opacus"})
```

## Definition of Done
Loading multiple libs in one go is possible with tests to show it works.

### PR DESCRIPTION
**Pull request by vsquareg**
## Description
Fix #5344
Enable loading multiple libraries in a single `syft.load` call.

## How has this been tested?
`tests/syft/api/load_lib_test.py`

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: src/syft/lib/__init__.py [hunk 1]
```diff
@@ -142,20 +145,26 @@ def _load_lib(*, lib: str, options: TypeDict[str, TypeAny] = {}) -> None:
...
-        _load_lib(lib=lib, options=options)
-    except VendorLibraryImportException as e:
-        critical(e)
-    except Exception as e:
-        critical(f"Unable to load package support for: {lib}. {e}")
+    if not isinstance(libs[0], str):
+        libs = tuple(libs[0])
+    for lib in libs:
+        try:
+            _load_lib(lib=str(lib), options=options)
```
Comment (gmuraru) [inline src/syft/lib/__init__.py:163]: Q: Why is the need to convert to string?

File: src/syft/lib/__init__.py [hunk 2]
```diff
@@ -171,7 +180,7 @@ def load_lib(lib: str, options: TypeDict[str, TypeAny] = {}) -> None:
     msg = "load_lib() is deprecated please use load() in the future"
     warning(msg, print=True)
     warnings.warn(msg, DeprecationWarning)
-    load(lib=lib, options=options)
```
Comment (gmuraru) [inline src/syft/lib/__init__.py:174]: NIT: Leave it like ```lib=lib```

File: src/syft/lib/__init__.py [hunk 3]
```diff
@@ -142,20 +145,26 @@ def _load_lib(*, lib: str, options: TypeDict[str, TypeAny] = {}) -> None:
             _regenerate_unions(lib_ast=lib_ast, client=client)
 
 
-def load(lib: str, options: TypeDict[str, TypeAny] = {}) -> None:
+def load(
```
Comment (tudorcebere) [inline src/syft/lib/__init__.py:148]: Could you add in the type hints that we want specifically strings?
`List[str], Set[str]` etc.

It has to be clear for the user that it shouldn't pass the module itself.

File: src/syft/lib/__init__.py [hunk 4]
```diff
@@ -142,20 +145,26 @@ def _load_lib(*, lib: str, options: TypeDict[str, TypeAny] = {}) -> None:
...
+        *libs: names of libraries to load and update Node with (can be variadic, tuple, list, set)
         options: external requirements for loading library successfully
     """
-    try:
-        _load_lib(lib=lib, options=options)
-    except VendorLibraryImportException as e:
-        critical(e)
-    except Exception as e:
-        critical(f"Unable to load package support for: {lib}. {e}")
+    if not isinstance(libs[0], str):
```
Comment (tudorcebere) [inline src/syft/lib/__init__.py:159]: We shouldn't convert to string, we need to check all the arguments (even if nested) are strings.

In other words:

```
if isintance(args[0], str):
   ...
elif is_iterable(args[0]):
   # iterate through this and check if the components are strings and load them
else:
   # error

File: tests/syft/api/load_lib_test.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# syft absolute
+import syft as sy
+
+
```
Comment (tudorcebere) [inline tests/syft/api/load_lib_test.py:4]: Could we get another test that checks for different errors when an invalid string or an invalid object is passed?

**Review by tudorcebere — CHANGES_REQUESTED** [review]
Really awesome work!

Added a few comments, you are on the right track!

File: src/syft/lib/__init__.py (hunk 1 above)
Comment (vsquareg) [reply src/syft/lib/__init__.py]: To ensure only string objects get passed.

Now checking using `isinstance` before calling `_load_lib`, to better handle error reporting.

File: src/syft/lib/__init__.py (hunk 2 above)
Comment (vsquareg) [reply src/syft/lib/__init__.py]: Done.

File: tests/syft/api/load_lib_test.py (hunk 5 above)
Comment (vsquareg) [reply tests/syft/api/load_lib_test.py:7]: Added error testing.

File: src/syft/lib/__init__.py (hunk 4 above)
Comment (vsquareg) [reply src/syft/lib/__init__.py]: Done.

File: src/syft/lib/__init__.py (hunk 3 above)
Comment (vsquareg) [reply src/syft/lib/__init__.py:149]: Done.

**Comment by vsquareg** [comment]
`test_load_errors` is testing the `sy.load` for errors raised within the logger for invalid calls.

This requires #5397 to be merged first.

**Review by tudorcebere — APPROVED** [review]
LGTM!

**Pull request merged**